### DISCUSSION TITLE
User configurable backup sync interval

### DISCUSSION THREAD

**Discussion by mikeynp** [lavinir/hassio-onedrive-backup#43] [Ideas]
Hi, I'd find it really useful to have a user configurable option for the backup (and/or recovery) sync interval, as opposed to the default 5 minutes. I understand the default is so that the sensors get updated relatively frequently and potential failed backups get uploaded, but as I'm only making a backup once every 24 hours, a much less frequent sync, e.g. 30 minutes, would be perfectly fine for me (though maybe with a sensible upper/lower limit just so its not syncing too fast/missing allowed sync times). Thanks!

**Comment by lavinir** [comment]
This was actually available originally in configuration and removed in later versions. I think it just opens the door to things not working as expected when interval is too high (as you mentioned with sensor updates and missed back ups)

What is the concern though? Are you worried about resource usage? Once every 5 minutes running for a few seconds with minimal cpu shouldn't take much of a toll.

**Reply by mikeynp** [reply]
Thanks for the reply. Perhaps you're right and I do get that you've got to cater for the majority, but main issue is that I run a Pi Hole and it spams my logs more than anything else. I also don't have an issue with the OneDrive server being contacted _only_ when a backup is due to take place - if it fails, so be it (you do get a notification this has happened) and any local backups will be uploaded at the next interval (or maybe it could retry the upload a few times in case of failure?). Maybe this setting could be one of the hidden configuration options for those who want to change it with [a] a warning that this might break things if changed from the default and [b] enforced sensible limits (if that is possible), e.g. greater than a minimum timeout, for example 1 minute,  and less that the configured "allowed hours" range or "backup interval"?
 
![1](https://user-images.githubusercontent.com/112195228/215147496-32b28ee4-37c7-44f5-9954-cc610d3f09e3.png)
